contributors/design-proposals/cloudprovider-storage-metrics.md

```diff
@@ -49,37 +49,87 @@ Since we are interested in count(or rate) and latency percentile metrics of API
 the external Cloud Provider - we will use [Histogram](https://prometheus.io/docs/practices/histograms/) type for
 emitting these metrics.
 
-We will be using `HistogramVec` type so as we can attach dimensions at runtime. Whenever available
-`namespace` will reported as a dimension with the metric.
+We will be using `HistogramVec` type so as we can attach dimensions at runtime. All metrics will container API action
+being taken as a dimension. The cloudprovider maintainer may choose to add additonal dimensions as needed.
 
-### GCE Implementation
+We are also interested in counter of cloudprovider API errors. `NewCounterVec` type will be used for keeping
+track of API errors.
 
-For GCE we simply use `gensupport.RegisterHook()` to register a function which will be called
-when request is made and response returns.
+### GCE Implementation
 
 To begin with we will start emitting following metrics for GCE. Because these metrics are of type
-`Summary` - both count and latency will be automatically calculated.
+`Histogram` - both count and latency will be automatically calculated.
+
+#### GCE Latency metrics
+
+All gce latency metrics will be named - `cloudprovider_gce_api_duration_seconds`. api request
```

Comment:
Now thinking about it, we may not want all of the metrics to be put into the same name, as this makes all of the metrics need to share the same dimensions (e.g. zone, region, etc). The dimensions may not make sense for most of the metrics.

For example, if you look at my PR, disk has (zone, node), but other metrics only have (zone) or ().

It probably would better to have:
cloudprovider_{gce,aws, ...}_{disk,loadbalancer, ...}

to avoid having too many unused properties.

What do you think?